nixos/tests/acme.nix

```diff
@@ -79,15 +108,48 @@ in import ./make-test-python.nix {
             '';
           };
         })
+        ({pkgs, config, nodes, lib, ...}: {
+          security.acme.certs."example.com" = {
+            domain = "*.example.com";
+            dnsProvider = "exec";
+            dnsPropagationCheck = false;
+            credentialsFile = with pkgs; writeText "wildcard.env" ''
+              EXEC_PATH=${dnsScript { inherit writeScript bash curl; dnsAddress = nodes.dnsserver.config.networking.primaryIPAddress; }}
+            '';
+            user = config.services.nginx.user;
+            group = config.services.nginx.group;
```

Comment:
Why is this necessary? Can't we use the per-vhost `enableACME` option in the nginx module, and just reconfigure the dns provider in the acme module?

Reply:
I'm testing wildcard certs. You can't specify a wildcard cert through enableACME, and that option is tested elsewhere. Personally I use the cert for more than just nginx so I want to ensure it works hard way.

Reply:
I'm not sure - I'd expect `serverAliases` to work with that - if not, I'd consider it a bug.

I just find this test hard to understand. - If we can't use the nginx-provided ACME functionalities, probably it'd be more understandable to just not use nginx in the test, only configure the acme module, and check test certificate files appear.

Reply:
I don't think it's sufficient to simply check does the file exist to truly test wildcard certs. It should work when used with any service (in this case, nginx) serving under that domain. Using nginx here keeps things consistent with the other tests. You're right I could've jused used `serverAliases` and added `*.example.com` there and it would've worked fine, but I'm testing the acme module, not nginx.

Reply:
I meant using nginx, but not the acme-specific support of it makes the test much more verbose. Maybe a nginx test with`serverAliases`, and a simple one which just looks at the certificate via `openssl x509 -in full.pem -text -noout | grep DNS`?